### PR TITLE
Added chunking mechanism to mda 

### PR DESCRIPTION
**Pull request by alejoe91**
This is done by directly writing the MdaHeader and appending the binary data using the `write_to_binary_dat_format` (which now can directly appen to a file handle)

### CODE REVIEW

File: spikeextractors/extraction_tools.py [hunk 1]
```diff
@@ -252,7 +252,8 @@ def read_binary(file, numchan, dtype, time_axis=0, offset=0):
     return samples
 
 
-def write_to_binary_dat_format(recording, save_path, time_axis=0, dtype=None, chunk_size=None, chunk_mb=500):
+def write_to_binary_dat_format(recording, save_path=None, file_handle=None,
+                               time_axis=0, dtype=None, chunk_size=None, chunk_mb=500):
```
Comment (samuelgarcia) [inline spikeextractors/extraction_tools.py]: **chunk_mb** very very cool. I will propagate into sorters.

Comment (alejoe91) [reply spikeextractors/extraction_tools.py]: by default it's 500mb. I think it's an ok value :)

File: spikeextractors/extractors/mdaextractors/mdaio.py [hunk 2]
```diff
@@ -4,214 +4,243 @@
...
     r = requests.get(url, headers=headers, stream=True)
     fd, tmp_fname = tempfile.mkstemp()
     with open(tmp_fname, 'wb') as f:
-        for chunk in r.iter_content(chunk_size=1024): 
+        for chunk in r.iter_content(chunk_size=1024):
             if chunk:
                 f.write(chunk)
     return tmp_fname
-        
+
```
Comment (samuelgarcia) [inline spikeextractors/extractors/mdaextractors/mdaio.py]: Really sure about this line ?

Comment (alejoe91) [reply spikeextractors/extractors/mdaextractors/mdaio.py]: extremely! Very important change

**Comment by samuelgarcia** [comment]
This PR looks OK.
Do you plan to write an test ?

**Comment by alejoe91** [comment]
The chunking is already called by default and it is used with the basic tests. The tests for the write_binary chunking are there already

**Comment by alejoe91** [comment]
@samuelgarcia can I merge? :)

**Pull request merged**